eclair-core/src/test/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisherSpec.scala

```diff
@@ -358,16 +409,196 @@ class ReplaceableTxPublisherSpec extends TestKitBaseClass with AnyFunSuiteLike w
       assert(result.tx.txIn.length > 2) // we added more than 1 wallet input
       assert(mempoolTxs.map(_.txid).contains(result.tx.txid))
-    })
+    }
+  }
+
+  test("commit tx fees not increased when confirmation target is far and feerate hasn't changed") {
+    withFixture(Seq(500 millibtc), ChannelTypes.AnchorOutputsZeroFeeHtlcTx) { f =>
+      import f._
+
+      val (commitTx, anchorTx) = closeChannelWithoutHtlcs(f, aliceBlockHeight() + 30)
+      wallet.publishTransaction(commitTx.tx).pipeTo(probe.ref)
+      probe.expectMsg(commitTx.tx.txid)
+
+      setFeerate(FeeratePerKw(3000 sat))
+      publisher ! Publish(probe.ref, anchorTx)
+      // wait for the commit tx and anchor tx to be published
+      val mempoolTxs = getMempoolTxs(2)
+      assert(mempoolTxs.map(_.txid).contains(commitTx.tx.txid))
+
+      // A new block is found, but we still have time and the feerate hasn't changed, so we don't bump the fees.
+      // Note that we don't generate blocks, so the transactions are still unconfirmed.
+      system.eventStream.publish(CurrentBlockCount(aliceBlockHeight().toLong + 5))
+      probe.expectNoMessage(500 millis)
+      val mempoolTxs2 = getMempool()
+      assert(mempoolTxs.map(_.txid).toSet === mempoolTxs2.map(_.txid).toSet)
+    }
+  }
+
+  test("commit tx not confirming, lowering anchor output amount") {
+    withFixture(Seq(500 millibtc), ChannelTypes.AnchorOutputsZeroFeeHtlcTx) { f =>
+      import f._
+
+      val (commitTx, anchorTx) = closeChannelWithoutHtlcs(f, aliceBlockHeight() + 30)
+      wallet.publishTransaction(commitTx.tx).pipeTo(probe.ref)
+      probe.expectMsg(commitTx.tx.txid)
+
+      val oldFeerate = FeeratePerKw(3000 sat)
+      setFeerate(oldFeerate)
+      publisher ! Publish(probe.ref, anchorTx)
+      // wait for the commit tx and anchor tx to be published
+      val mempoolTxs1 = getMempoolTxs(2)
+      assert(mempoolTxs1.map(_.txid).contains(commitTx.tx.txid))
+      val mempoolAnchorTx1 = mempoolTxs1.filter(_.txid != commitTx.tx.txid).head
+
+      // A new block is found, and the feerate has increased for our block target, so we bump the fees.
+      val newFeerate = FeeratePerKw(5000 sat)
+      setFeerate(newFeerate, blockTarget = 12)
+      system.eventStream.publish(CurrentBlockCount(aliceBlockHeight().toLong + 5))
+      awaitCond(!isInMempool(mempoolAnchorTx1.txid), interval = 200 millis, max = 30 seconds)
+      val mempoolTxs2 = getMempoolTxs(2)
+      val mempoolAnchorTx2 = mempoolTxs2.filter(_.txid != commitTx.tx.txid).head
+      assert(mempoolAnchorTx1.fees < mempoolAnchorTx2.fees)
+
+      val targetFee = Transactions.weight2fee(newFeerate, mempoolTxs2.map(_.weight).sum.toInt)
+      val actualFee = mempoolTxs2.map(_.fees).sum
+      assert(targetFee * 0.9 <= actualFee && actualFee <= targetFee * 1.1, s"actualFee=$actualFee targetFee=$targetFee")
+    }
+  }
+
+  test("commit tx not confirming, adding other wallet inputs") {
```

Comment:
Just to be sure: in this test we actually replace the first wallet input by the second wallet input. Would it make sense to test the case where we use both inputs?

Reply:
Can you detail what you mean by `use both inputs`? At what step exactly? During the retry or in the first attempt?

Reply:
IIUC, we have two available inputs in our wallet: 10.5 mbtc and 5 mbtc and we are increasing the feerate to make sure we are able to use up more of our available inputs as indeed. Maybe we could construct the test such that we first use the 5 mbtc input, then the 10.5 mbtc input, then both inputs?

Reply:
The thing is we can never use both.
While txA is in the mempool (spending inputA), txB can only use inputB for its funding (it cannot steal inputA from txA).
But do you mean that we should add a new attempt that would need both inputs to reach the feerate and thus will fail?

Reply:
> While txA is in the mempool (spending inputA), txB can only use inputB for its funding (it cannot steal inputA from txA).

I didn't realize that! How about adding a third input and another fee bump, to make sure that for we then use two inputs (the 1st that has been unlocked and the yet-unused one)?

Reply:
That's really hard to do since we don't control bitcoind's coin selection...

Reply:
Fine let's leave it as-is.